File: workflowtests/python/tests/simple-test/src/worker.py
from omes_starter import WorkerConfig
from temporalio.worker import Worker

from .workflows import SimpleWorkflow


async def worker_main(config: WorkerConfig) -> Worker:
    """Configure and return the worker."""
    return Worker(
        config.client,
        task_queue=config.task_queue,
        workflows=[SimpleWorkflow],
    )

File: workflowtests/typescript/tests/simple-test/src/worker.ts
import { NativeConnection, Worker, bundleWorkflowCode } from '@temporalio/worker';
import type { WorkerConfig } from '@temporalio/omes-starter';

/**
 * Configure and return the worker.
 */
export async function workerMain(config: WorkerConfig): Promise<Worker> {
    const workflowsPath = require.resolve('./workflows');
    const workflowBundle = await bundleWorkflowCode({ workflowsPath });

    const nativeConnection = await NativeConnection.connect(config.connectionOptions);

    return Worker.create({
        connection: nativeConnection,
        namespace: config.namespace,
        taskQueue: config.taskQueue,
        workflowBundle,
    });
}
